packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -1786,33 +1785,25 @@ export abstract class LanguageServerBase implements LanguageServerInterface, Dis
             const code = this.getDiagCode(diag, rule);
             const vsDiag = Diagnostic.create(diag.range, diag.message, severity, code, this.serverOptions.productName);
 
-            if (diag.category === DiagnosticCategory.Hint) {
-                if (diag.baselined) {
-                    vsDiag.message = `Baselined: ${vsDiag.message}`;
-                }
-                if (deprecatedDiagnosticRules().includes(rule)) {
-                    // If the client doesn't support "deprecated" tags, don't report deprecated diagnostics unless it's a baselined error.
-                    if (!this.client.supportsDeprecatedDiagnosticTag && !diag.baselined) {
-                        return;
-                    }
-                    vsDiag.tags = [DiagnosticTag.Deprecated];
-                } else if ([...unreachableDiagnosticRules(), ...unusedDiagnosticRules()].includes(rule)) {
-                    // If the client doesn't support "unnecessary" tags, don't report unused code unless it's a baselined error.
-                    if (!this.client.supportsUnnecessaryDiagnosticTag && !diag.baselined) {
-                        return;
-                    }
-                    vsDiag.tags = [DiagnosticTag.Unnecessary];
+            if (diag.category === DiagnosticCategory.TaskItem) {
+                // TaskItem is not supported.
+                return;
+            }
+            if (diag.baselined) {
+                vsDiag.message = `Baselined: ${vsDiag.message}`;
+            }
+            if (deprecatedDiagnosticRules().includes(rule)) {
+                // If the client doesn't support "deprecated" tags, don't report deprecated diagnostics unless it's a baselined error.
+                if (!this.client.supportsDeprecatedDiagnosticTag && !diag.baselined) {
+                    return;
```

Comment:
this makes it so that diagnostics that can have the tag are not reported at all if the client doesn't support them, even if the diagnostic is configured to be reported as a warning or an error.

i think when i did this i assumed diagnostic tags could only be applied to hint diagnostic but that's not the case. i'll fix this